src/openlcb/BroadcastTimeAlarm.hxx

```diff
@@ -172,24 +172,31 @@ private:
         /// @param alarm our parent alarm that we will awaken
         Wakeup(BroadcastTimeAlarm *alarm)
             : alarm_(alarm)
+            , armed(false)
         {
         }
 
         /// Trigger the wakeup to run.
         void trigger()
         {
-            alarm_->service()->executor()->add(this);
+            if (!armed)
```

Comment:
should `armed` be an atomic_bool or otherwise protected from concurrency?

Reply:
Fixed.